File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        
        int maxSum=0;
        int tempSum=0;
        
        for(int i=0; i<k; i++) {
            tempSum+=nums[i];
        }
        
        maxSum = tempSum;
        
        for(int j=k; j<nums.length;j++) {
            
            tempSum = tempSum-nums[j-k]+nums[j];
            
            maxSum = Math.max(maxSum, tempSum);
        }
        
        return maxSum/1.0/k;
        
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        temp_sum=0
        
        max_sum=-float(inf)
        i=0
        
        
        while i<len(nums) :
            
            while i<k:
            
                temp_sum+=nums[i]
                i+=1
                
            # print(max_sum)
            
            max_sum = max(max_sum, temp_sum)
            
            if i<len(nums):
                temp_sum = temp_sum - nums[i-k] + nums[i]

                max_sum = max(max_sum, temp_sum)
            i+=1
            # print(max_sum)
        
        return max_sum/k
            
            
        